File: web/js/planet.js
class Planet {
    constructor(name, x, y, radius, color, resources) {
        this.name = name;
        this.x = x;
        this.y = y;
        this.radius = radius;
        this.color = color;
        this.resources = resources || {};
    }

    draw(ctx) {
        // Градиент для планеты
        const gradient = ctx.createRadialGradient(
            this.x - this.radius * 0.3,
            this.y - this.radius * 0.3,
            0,
            this.x,
            this.y,
            this.radius
        );
        gradient.addColorStop(0, this.lightenColor(this.color, 30));
        gradient.addColorStop(1, this.color);
        
        // Основная планета
        ctx.fillStyle = gradient;
        ctx.beginPath();
        ctx.arc(this.x, this.y, this.radius, 0, Math.PI * 2);
        ctx.fill();
        
        // Обводка
        ctx.strokeStyle = 'rgba(255, 255, 255, 0.3)';
        ctx.lineWidth = 2;
        ctx.stroke();
        
        // Название
        ctx.fillStyle = '#ffffff';
        ctx.font = '16px Segoe UI';
        ctx.textAlign = 'center';
        ctx.fillText(this.name, this.x, this.y + this.radius + 20);
    }

    lightenColor(color, percent) {
        // Простая функция осветления цвета
        return color;
    }

    containsPoint(x, y) {
        const dx = x - this.x;
        const dy = y - this.y;
        return Math.sqrt(dx * dx + dy * dy) <= this.radius;
    }
}

File: src/entities/planet.py
"""
Планета - объект в космосе с ресурсами
"""

import pygame


class Planet:
    def __init__(self, name, x, y, radius, color, resources=None):
        self.name = name
        self.x = x
        self.y = y
        self.radius = radius
        self.color = color
        self.resources = resources or {}
        
    def draw(self, screen):
        """Отрисовка планеты"""
        # Основной круг
        pygame.draw.circle(
            screen,
            self.color,
            (int(self.x), int(self.y)),
            self.radius
        )
        
        # Обводка
        pygame.draw.circle(
            screen,
            (255, 255, 255),
            (int(self.x), int(self.y)),
            self.radius,
            2
        )
        
        # Название
        font = pygame.font.Font(None, 24)
        text = font.render(self.name, True, (255, 255, 255))
        text_rect = text.get_rect(center=(self.x, self.y + self.radius + 15))
        screen.blit(text, text_rect)
